Populate profile screen

ClaireCassidy opened the pull request:
## Purpose

We need to populate the user's profile screen with their saved creations, and give them a way to view their creations again fullscreen from within the app. We also want the user to be able to name their creations as a personalisation feature.

## Proposed Change

The profile screen uses ``Expo Media Library`` to pull saved user creations from the local ``/Paint-Yourself`` folder and populates the image grid.

User enters a name when they're saving their creations. Upon saving, a new {uri: creationName} pair is appended the async storage object mapping creations to their names. When the user selects an image from the grid on the profile screen, a modal with title ``creationName`` is opened with a fullscreen view of their image.

![image](https://user-images.githubusercontent.com/20342363/156372361-7373baf6-763e-4cdb-839d-f376497cc39f.png)

![image](https://user-images.githubusercontent.com/20342363/156372397-1f3ffe55-611b-49ee-81e1-a50e0b9beaf1.png)

![image](https://user-images.githubusercontent.com/20342363/156372438-ec02742b-9259-4746-8122-4708f5dd6371.png)


## Checklist

- [x] Permissions checks for media library & error handling
  - [x] Popup modal for when user has declined permissions ("please enable in settings")
- [x] Load creations from ``/Paint-Yourself`` gallery folder to profile screen grid
  - [x] Alternate view for when user has no creations
- [x] Set up async storage to handle saving pairs of URI and creationName
- [x] Add open Modal for onPress image grid item, showing screen for creation

ClaireCassidy committed:
check media permissions and modal if not granted

ClaireCassidy committed:
fetch creations from album into memory and styling

ClaireCassidy committed:
can display creations; some logic bugs to fix

ClaireCassidy committed:
fix initial image load logic and add reload when numCreations changes

ClaireCassidy committed:
add pull to refresh to creations container

ClaireCassidy committed:
add view for when user has no creations

ClaireCassidy committed:
fix bugs with refresh and returning to home screen too quickly after saving

ClaireCassidy committed:
tweaks to image grid styling

ClaireCassidy committed:
add name input and logic for saving; need to hook up to asyncstorage

ClaireCassidy committed:
create async storage object where doesn't exist

ClaireCassidy committed:
append new pairs to async storage object

ClaireCassidy committed:
add open expanded image modal screen (skeleton) on image grid

ClaireCassidy committed:
add route params for expanded image modal

ClaireCassidy committed:
modal shows creation's saved name

ClaireCassidy committed:
show image on expanded iamge modal page

ClaireCassidy committed:
ui tweaks for when permissions fail

DanielNugent reviewed (approved):
Looks great!

I think cause I changed the save button to an icon, you can feel free to revert it to the button or add a model where they can add the image file name when they press the save icon

ClaireCassidy commented:
> Looks great!
> 
> I think cause I changed the save button to an icon, you can feel free to revert it to the button or add a model where they can add the image file name when they press the save icon



> Looks great!
> 
> I think cause I changed the save button to an icon, you can feel free to revert it to the button or add a model where they can add the image file name when they press the save icon

Awesome, think for now we might just stick with mine where its all on one page for time's sake? Do like the look of the icon for the future tho.

lexesj reviewed (approved):
LGTM!
